docs/features/string-literals-data-section.md

```diff
@@ -0,0 +1,257 @@
+When the limit is reached, the following compiler error is reported by Roslyn:
+
+```
+error CS8103: Combined length of user strings used by the program exceeds allowed limit. Try to decrease use of string literals.
+```
+
+By turning on [the feature flag](#configuration), string literals ([where possible](#eligible-string-literals)) are instead emitted as UTF-8 data into a different section of the PE file
+which does not have the same limit. The emit format is similar to [explicit u8 string literals][u8-literals].
+
+The feature is currently implemented only for C#, not VB.
+
+> [!WARNING]
+> This feature is currently experimental and can be changed or removed at any time.
+
+## Configuration
+
+The feature flag can take a non-negative integer threshold.
+Only string literals whose length (number of characters, not bytes) is greater than the threshold are emitted using the utf8 encoding strategy.
+If the flag is set, but no value or an unrecognized value is specified, the threshold defaults to 100.
+Specifying 0 means all non-empty string literals are considered for the feature.
+
+The feature is turned off if
+- the feature flag is not specified, or
+- the string `off` is provided as the feature flag value, or
+- `null` is provided as the feature flag value (usually only possible using Roslyn APIs like `ParseOptions.WithFeatures`).
+
+The feature flag can be specified on the command line like `/features:experimental-data-section-string-literals` or `/features:experimental-data-section-string-literals=20`,
+or in a project file in a `<PropertyGroup>` like `<Features>$(Features);experimental-data-section-string-literals</Features>` or `<Features>$(Features);experimental-data-section-string-literals=20</Features>`.
+
+> [!NOTE]
+> This configuration is useful for experimenting (to see the impact of the threshold on runtime performance and IL size)
+> and also allowing users to avoid a potential overhead of many type definitions generated if their program has many short string literals
+> which do not need this new emit strategy because they can fit into the size-limited UserString heap.
+
+## Eligible string literals
+
+A string literal is emitted with the utf8 encoding if and only if all of these are true:
+- it can be encoded as UTF-8, i.e., the following call does not throw:
+  ```cs
+  new System.Text.Encoding.UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true).GetBytes(literal)
+  ```
+- its `Length` is greater than the threshold specified by [the feature flag](#configuration).
+
+## Emit
+
+The standard string literal emit strategy emits `ldstr "string"`.
+That puts the string literal on the UserString heap of the PE file.
+
+The utf8 string literal encoding emit strategy emits `ldsfld` of a field in a generated class instead.
+
+For every string literal, a unique internal static class is generated which:
```

Comment:
Should this be "For every **unique** string literal, an internal static class ..."?